File: frontend/sdk/core/iotex-client.js
import { Web3 } from 'web3';

export class IoTeXClient {
  constructor(config = {}) {
    this.rpcUrl = config.rpcUrl || 'https://babel-api.mainnet.iotex.io';
    this.web3 = new Web3(this.rpcUrl);
    this.chainId = 4689;
  }

  async connectWallet() {
    if (typeof window !== 'undefined' && window.ethereum) {
      try {
        await window.ethereum.request({ method: 'eth_requestAccounts' });
        const accounts = await this.web3.eth.getAccounts();
        return accounts[0];
      } catch (error) {
        throw new Error('Failed to connect wallet: ' + error.message);
      }
    }
    throw new Error('No Web3 wallet found');
  }

  async getBalance(address) {
    const balance = await this.web3.eth.getBalance(address);
    return this.web3.utils.fromWei(balance, 'ether');
  }

  async createEscrow(betData, amount) {
    const contract = new this.web3.eth.Contract(ESCROW_ABI, ESCROW_CONTRACT_ADDRESS);
    const accounts = await this.web3.eth.getAccounts();
    return contract.methods.createBet(betData).send({
      from: accounts[0],
      value: this.web3.utils.toWei(amount, 'ether')
    });
  }
}

File: geoprop-ai/backend/blockchain/iotex_client.py
from web3 import Web3 
import asyncio 
import json 
import os 
from typing import Dict, Any, Optional 
import httpx 
 
class IoTeXClient: 
    def __init__(self): 
        self.rpc_url = os.getenv("IOTEX_RPC_URL", "https://babel-api.mainnet.iotex.io") 
        self.web3 = Web3(Web3.HTTPProvider(self.rpc_url)) 
        self.chain_id = 4689 
        self.realms_api = "https://api.iotex.io/realms/v1" 
 
    async def get_iotx_balance(self, address: str) -
        try: 
            balance_wei = self.web3.eth.get_balance(address) 
            return self.web3.from_wei(balance_wei, 'ether') 
        except Exception as e: 
            raise Exception(f"Failed to get IOTX balance: {e}") 
